### PR TITLE
Added a couple of simple LM examples and a scikit-sparse/cholesky decomposition example

### PR DESCRIPTION
**Pull request by krlberry**

### CODE REVIEW

File: examples/LM_Bundle_Adjust.ipynb [hunk 1]
```diff
@@ -0,0 +1,725 @@
...
+    "iteration = 0\n",
+    "V = compute_residuals(cnet, sensors)\n",
+    "dX = np.zeros(W_params.shape[0]) #initialize for sigma calculatio\n",
+    "sigma0 = compute_sigma(V, dX, W_params, W_observations)\n",
+    "print(f'iteration {iteration}: sigma0 = {sigma0}\\n')\n",
+    "\n",
+    "max_iterations = 100\n",
+    "tol = 1e-10\n",
+    "total_correction = np.zeros(num_parameters)\n",
+    "damping = 100\n",
```
Comment (ladoramkershner) [inline examples/LM_Bundle_Adjust.ipynb]: Earlier you state that the Numerical Recipies's heuristic was damping = 0.01 and I am not seeing 1/damping in the Normal Equation below.

File: examples/LM_Bundle_Adjust.ipynb [hunk 2]
```diff
@@ -0,0 +1,725 @@
...
+   "metadata": {},
+   "source": [
+    "## Whole (Dense) Bundle Process in a loop with LM terms\n",
+    "\n",
+    "The next three cells show examples of the bundle adjustment loop running with Levenberg-Marquat added.\n",
+    "\n",
+    "This example shows the first heuristic for initializing $\\lambda$, the weighting parameter, from Numerical Recipes in C++\n",
+    "\n",
+    "NumericalRecipes in C++ recommended starting with $\\lambda$ = 0.01, and then increasing or decreasing this value by an order of magnitude each iteration depending on whether the last iteration increased or decreased \n",
+    "$\\sigma_0$.\n",
```
Comment (ladoramkershner) [inline examples/LM_Bundle_Adjust.ipynb]: Do you have a link for where they suggest this?

File: examples/LM_Bundle_Adjust.ipynb [hunk 3]
```diff
@@ -0,0 +1,725 @@
...
+    "max_iterations = 100\n",
+    "tol = 1e-10\n",
+    "total_correction = np.zeros(num_parameters)\n",
+    "damping = 100\n",
+    "for i in range(max_iterations):   \n",
+    "    iteration += 1\n",
+    "    old_sigma0 = sigma0\n",
+    "    \n",
+    "    J = compute_jacobian(cnet, sensors, parameters, column_dict)    \n",
+    "    C = J.T.dot(W_observations).dot(V) - damping*np.identity(W_params.shape[0]).dot(total_correction) - W_params.dot(total_correction)\n",
```
Comment (ladoramkershner) [inline examples/LM_Bundle_Adjust.ipynb]: I am be wrong, but looking at Wikipedia and the paper you cited below (for the sba/levmar's heuristic) I am not seeing the dampening term being subtracted to the C term, only added to the normal equation.

File: examples/LM_Bundle_Adjust.ipynb [hunk 4]
```diff
@@ -0,0 +1,725 @@
...
+    "damping = 0.001\n",
+    "for i in range(max_iterations):   \n",
+    "    iteration += 1\n",
+    "    old_sigma0 = sigma0\n",
+    "    \n",
+    "    J = compute_jacobian(cnet, sensors, parameters, column_dict)\n",
+    "    if i==0:\n",
+    "        damping *= np.max(J.T.dot(J))\n",
+    "    \n",
+    "    C = J.T.dot(W_observations).dot(V) - damping*np.identity(W_params.shape[0]).dot(total_correction) - W_params.dot(total_correction)\n",
```
Comment (ladoramkershner) [inline examples/LM_Bundle_Adjust.ipynb]: Same thing as comment above, does dampening coefficient get applied to C?

File: examples/LM_Bundle_Adjust.ipynb [hunk 5]
```diff
@@ -0,0 +1,725 @@
...
+    "    old_V = V\n",
+    "    V = compute_residuals(cnet, sensors)\n",
+    "    sigma0 = compute_sigma(V, dX, W_params, W_observations)\n",
+    "    if (sigma0 < old_sigma0):\n",
+    "        damping = max(1.0/3.0, (np.linalg.norm(old_V)**2 - np.linalg.norm(V)**2)/(dX.T.dot(damping*dX + C)))\n",
+    "        nu = 2\n",
+    "        print(\"DAMPING1:\", damping, \"NU1:\", nu)\n",
+    "    else:\n",
+    "        damping = damping * nu\n",
+    "        nu = 2 * nu\n",
```
Comment (ladoramkershner) [inline examples/LM_Bundle_Adjust.ipynb]: What happens if sigma increases and there is not a nu set? Should nu be set outside of the main loop?

File: examples/LM_Bundle_Adjust.ipynb (hunk 2 above)
Comment (krlberry) [reply examples/LM_Bundle_Adjust.ipynb]: I responded via teams internally. There isn't an online source for this.

File: examples/LM_Bundle_Adjust.ipynb (hunk 1 above)
Comment (krlberry) [reply examples/LM_Bundle_Adjust.ipynb]: Oops. I was playing around with different values to see how it affected the convergance and clearly forgot to revert this before PRing in. Thanks for noticing this!

File: examples/LM_Bundle_Adjust.ipynb (hunk 2 above)
Comment (ladoramkershner) [reply examples/LM_Bundle_Adjust.ipynb]: I was mainly suggesting it for documentation purposes, but based on the conversation we had earlier.. @jessemapel @kberryUSGS should we maybe not have this method in the notebook in the repo?

File: examples/LM_Bundle_Adjust.ipynb (hunk 3 above)
Comment (krlberry) [reply examples/LM_Bundle_Adjust.ipynb]: Oops, you're probably right. Thanks for catching this!

File: examples/LM_Bundle_Adjust.ipynb (hunk 5 above)
Comment (krlberry) [reply examples/LM_Bundle_Adjust.ipynb]: Probably.

**Comment by jessemapel** [comment]
This are purely demo notebooks so I am going to merge them into the repo.

**Comment by jessemapel** [comment]
This are purely demo notebooks so I am going to merge them into the repo.

**Pull request merged**